src/neural/onnx/converter.cc

```diff
@@ -186,12 +236,121 @@ void Converter::AddStdInitializers(OnnxBuilder* builder) {
                           Int64OnnxConst({-1, NumFilters() * 2, 1, 1}, {4}));
 }
 
+std::string Converter::MakeEncoderLayer(
+    OnnxBuilder* builder, const LegacyWeights::EncoderLayer& layer,
+    int embedding_size, int heads, const std::string& encoder_in,
+    const std::string& name) {
+  const int d_model = layer.mha.q_b.size();
+  const int depth = d_model / heads;
+
+  auto mha_shape =
+      builder->AddInitializer("/const" + name + "/mha/shape",
+                              Int64OnnxConst({-1, 64, heads, depth}, {4}));
+
+  auto flow = builder->MatMul(
+      name + "/mha/Q/w", encoder_in,
+      *GetWeghtsConverter(layer.mha.q_w, {embedding_size, d_model}, {1, 0}));
+  flow = builder->Add(name + "/mha/Q/b", flow,
+                      *GetWeghtsConverter(layer.mha.q_b, {d_model}));
+
+  flow = builder->Reshape(name + "/mha/Q/reshape", flow, mha_shape);
+
+  auto Q = builder->Transpose(name + "/mha/Q/transpose", flow, {0, 2, 1, 3});
+
+  flow = builder->MatMul(
+      name + "/mha/K/w", encoder_in,
+      *GetWeghtsConverter(layer.mha.k_w, {embedding_size, d_model}, {1, 0}));
+  flow = builder->Add(name + "/mha/K/b", flow,
+                      *GetWeghtsConverter(layer.mha.k_b, {d_model}));
+
+  flow = builder->Reshape(name + "/mha/K/reshape", flow, mha_shape);
+
```

Comment:
Just a common impression, this function looks too sparse. :)
With so many empty lines.